packages/utils/src/totals/index.ts

```diff
@@ -0,0 +1,250 @@
+import {
+  BigNumberRawValue,
+  CartDTO,
+  CartShippingMethodDTO,
+} from "@medusajs/types"
+import { BigNumber as BigNumberJs } from "bignumber.js"
+import { BigNumber } from "./big-number"
+
+type GetLineItemTotalsContext = {
+  includeTax?: boolean
+  taxRate?: number | null
+}
+
+interface GetShippingMethodTotalInput extends CartShippingMethodDTO {
+  raw_amount: BigNumberRawValue
+}
+
+interface GetItemTotalInput {
+  id: string
+  unit_price: BigNumber
+  quantity: number
+  is_tax_inclusive?: boolean
+  tax_total?: BigNumber
+  original_tax_total?: BigNumber
+}
+
+interface GetItemTotalOutput {
+  quantity: number
+  unit_price: BigNumber
+
+  subtotal: BigNumber
+  total: BigNumber
+  original_total: BigNumber
+  discount_total: BigNumber
+  tax_total: BigNumber
+  original_tax_total: BigNumber
+}
+
+export function getShippingMethodTotals(
+  shippingMethods: GetShippingMethodTotalInput[],
+  context: { includeTax?: boolean }
+) {
+  const { includeTax } = context
+
+  const shippingMethodsTotals = {}
+
+  for (const shippingMethod of shippingMethods) {
+    shippingMethodsTotals[shippingMethod.id] = getShippingMethodTotals_(
+      shippingMethod,
+      {
+        includeTax,
+      }
+    )
+  }
+
+  return shippingMethodsTotals
+}
+
+export function getShippingMethodTotals_(
```

Comment:
comment: This is the old approach. Please look at `getTotalsForSingleLineItem` for an update one.